### ISSUE TITLE
How to report unnecessary `pyright: ignore` comments, but not `type: ignore` ?

### ISSUE THREAD

**Issue by Avasam** [microsoft/pyright#4243]
**Is your feature request related to a problem? Please describe.**
I would like pyright to report unecessary `pyright: ignore` comments. But completely leave alone `type: ignore` in libraries I'm validating against both mypy and pyright.

The following config still reports `Unnecessary "# type: ignore" comment`
```jsonc
"reportUnnecessaryTypeIgnoreComment": "error",
// Leave `type: ignore` comments to mypy
"enableTypeIgnoreComments": false,
```
With `"enableTypeIgnoreComments": true`, any `# type: ignore[whatever]` will disable all pyright errors for that line. And will also report if added to a line with a mypy-only error-code.

**Describe the solution you'd like**
A way for pyright to stop checking `# type: ignore` comments completely. (but not `pyright: ignore`)

**Comment by erictraut** [comment]
I'm reluctant to add such a feature because it's effectively working around a limitation in mypy. Mypy should provide its own way of [suppressing diagnostics](https://github.com/python/mypy/issues/12358) (a `# mypy: ignore` comment).

I think it's reasonable to suppress the `reportUnnecessaryTypeIgnoreComment` diagnostics if `enableTypeIgnoreComments` is set to false. That gives you the option of disabling `enableTypeIgnoreComments` entirely, treating all `# type: ignore` as though they didn't exist. I've made that change, and it will be included in the next release.

**Comment by Avasam** [comment]
I agree about mypy and I've upvoted the issue you linked.

Thank you!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.282, which I just published. It will also be included in a future release of pylance.

**Comment by Avasam** [comment]
Thanks! It mostly work but there's still 1 edge case: if a mypy and pyright error *do* happen on the same line. I can't ignore both because both expect their ignore to be at the start of the comment.
Same line doesn't even necessarily mean same error. But certain formatters (like black), will force everytihing on a line until the set limit.

Example:
```py
class Task(Future[_T_co], Generic[_T_co]):  # type: ignore[type-var]  # pyright: ignore[reportGeneralTypeIssues]
  # [...]
  pass
 
// OR
class Task(Future[_T_co], Generic[_T_co]):  # pyright: ignore[reportGeneralTypeIssues]  # type: ignore[type-var]
  # [...]
  pass
```

And I have a hunch due to discussions in # 12358 that mypy isn't the one that's gonna be able to support ignores at the end of the line.
This could also be an issue with other tools that want their special keywords at the start of the line.

**Comment by Avasam** [comment]
Alternatively a `pyright: disable-next-line` or block disable/enable would also work.

**Comment by erictraut** [comment]
I consider this ticket closed. If you have additional feature requests, please open a new ticket.

We're unlikely to implement "next line" or block disable. That has been suggested before and has been rejected. You could search for that issue and upvote it. We sometimes reconsider decisions if a closed issue has been upvoted by many pyright users.
